src/scriptrag/api/database_operations.py

```diff
@@ -0,0 +1,499 @@
+
+    def clear_scene_content(self, conn: sqlite3.Connection, scene_id: int) -> None:
+        """Clear dialogues and actions for a scene before re-inserting.
+
+        Args:
+            conn: Database connection
+            scene_id: ID of the scene to clear
+        """
+        conn.execute("DELETE FROM dialogues WHERE scene_id = ?", (scene_id,))
+        conn.execute("DELETE FROM actions WHERE scene_id = ?", (scene_id,))
+
+    def insert_dialogues(
+        self,
+        conn: sqlite3.Connection,
+        scene_id: int,
+        dialogues: list[Dialogue],
+        character_map: dict[str, int],
+    ) -> int:
+        """Insert dialogue records.
+
+        Args:
+            conn: Database connection
+            scene_id: ID of the parent scene
+            dialogues: List of dialogue objects
+            character_map: Mapping of character names to IDs
+
+        Returns:
+            Number of dialogues inserted
+        """
+        count = 0
+        for order, dialogue in enumerate(dialogues):
+            if dialogue.character not in character_map:
+                # Skip dialogues for unknown characters
+                logger.warning(f"Unknown character in dialogue: {dialogue.character}")
+                continue
+
+            metadata = {}
+            if dialogue.parenthetical:
+                metadata["parenthetical"] = dialogue.parenthetical
+
+            conn.execute(
+                """
+                INSERT INTO dialogues (scene_id, character_id, dialogue_text,
+                                     order_in_scene, metadata)
+                VALUES (?, ?, ?, ?, ?)
+                """,
+                (
+                    scene_id,
+                    character_map[dialogue.character],
+                    dialogue.text,
+                    order,
+                    json.dumps(metadata) if metadata else None,
+                ),
+            )
+            count += 1
+
+        logger.debug(f"Inserted {count} dialogues for scene {scene_id}")
+        return count
+
+    def insert_actions(
+        self, conn: sqlite3.Connection, scene_id: int, actions: list[str]
+    ) -> int:
+        """Insert action records.
+
+        Args:
+            conn: Database connection
+            scene_id: ID of the parent scene
+            actions: List of action text lines
+
+        Returns:
+            Number of actions inserted
+        """
+        count = 0
+        for order, action_text in enumerate(actions):
+            if not action_text.strip():
+                continue
+
+            conn.execute(
+                """
+                INSERT INTO actions (scene_id, action_text, order_in_scene)
+                VALUES (?, ?, ?)
+                """,
+                (scene_id, action_text, order),
+            )
+            count += 1
+
+        logger.debug(f"Inserted {count} actions for scene {scene_id}")
+        return count
+
+    def _extract_location(self, heading: str) -> str | None:
```

Comment:
This looks like code duplication. For the scriptrag list and scriptrag analyze command we likely have something like this already and if not, then putting this into a ScreenplayUtils class instead of burying it in the database code would be better.